packages/nextlove/src/with-route-spec/middlewares/with-validation.ts

```diff
@@ -114,14 +119,56 @@ const zodIssueToString = (issue: z.ZodIssue) => {
   return `${issue.message} for "${issue.path.join(".")}"`
 }
 
+function validateJsonResponse<JsonResponse extends z.ZodTypeAny>(
```

Comment:
Function should do as it's named

```suggestion
function conditionallyValidateJsonResponse<JsonResponse extends z.ZodTypeAny>(
```

a better alternative to this is to have `shouldValidateResponses` as a separate function, then in the middleware do

```ts
if (shouldValidateResponses()) {
  validateJsonResponse(...)
}
```